app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -48,8 +48,11 @@ def set_bookmark
 
   # Never trust parameters from the scary internet, only allow the white list through.
   def bookmark_params
-    params.require(:data).require(:attributes).permit(:title, :description).tap do |obj|
-      obj[:tags] = Tag.find_or_create_tags tags: params.dig(:data, :attributes).fetch(:tags, [])
+    params.require(:data).require(:attributes).permit(:title, :description, :tags).tap do |obj|
+      tags = Tag.find_or_create_tags tags: params.dig(:data, :attributes).fetch(:tags, [])
```

Comment:
do you really want to be doing this in your params method? Seems like you could accidentally call this code more than once. I'm actually a little surprised you aren't already.

Reply:
I'm afraid of having an object named "BookmarkUpserter"